Use `IEnumerable` in service will generated invalid formatter

### Description

Create a service as [follows](https://github.com/chika-chang/MagicOnionFormattersIssueSample/blob/main/Sample.Unity/Assets/Scripts/ServerShared/Services/ISampleService.cs#L10-L21):

```cs
public interface ISampleService : IService<ISampleService>
{
    UnaryResult<ICollection<int>> GetCollectionAsync();
    UnaryResult<IEnumerable<int>> GetEnumerableAsync();
    UnaryResult<IDictionary<int, int>> GetDictionaryAsync();
    UnaryResult<IGrouping<int, int>> GetGroupAsync();
    UnaryResult<IList<int>> GetListAsync();
    UnaryResult<ILookup<int, int>> GetLookupAsync();
    UnaryResult<IReadOnlyCollection<int>> GetReadOnlyCollectionAsync();
    UnaryResult<IReadOnlyDictionary<int, int>> GetReadOnlyDictionaryAsync();
    UnaryResult<IReadOnlyList<int>> GetReadOnlyListAsync();
}
```

### Expected Behavior

All clean, everything is happy.

### Current Behavior

Unity console as follows:

```
Assets\Scripts\Generated\MagicOnion.Generated.cs(107,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(108,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(109,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(110,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(111,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(112,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(113,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(114,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
Assets\Scripts\Generated\MagicOnion.Generated.cs(115,67): error CS0234: The type or namespace name 'System' does not exist in the namespace 'MessagePack.Formatters' (are you missing an assembly reference?)
```

### Sample repository

[MagicOnionFormattersIssueSample](https://github.com/chika-chang/MagicOnionFormattersIssueSample.git)

### Environment

- MagicOnion: 4.5.0
- MessagePack: 2.4.35
- Unity: 2021.3.5f1
- Opeartion System: Windows 11 21H2

--- a/src/MagicOnion.GeneratorCore/MagicOnionCompiler.cs
+++ b/src/MagicOnion.GeneratorCore/MagicOnionCompiler.cs
@@ -282,7 +282,16 @@ namespace MagicOnion
         static readonly Dictionary<string, string> additionalSupportGenericFormatter = new Dictionary<string, string>
         {
             {"System.Collections.Generic.List<>", "global::MessagePack.Formatters.ListFormatter<TREPLACE>()" },
+            {"System.Collections.Generic.IList<>", "global::MessagePack.Formatters.InterfaceListFormatter2<TREPLACE>()" },
+            {"System.Collections.Generic.IReadOnlyList<>", "global::MessagePack.Formatters.InterfaceReadOnlyListFormatter<TREPLACE>()" },
             {"System.Collections.Generic.Dictionary<,>", "global::MessagePack.Formatters.DictionaryFormatter<TREPLACE>()"},
+            {"System.Collections.Generic.IDictionary<,>", "global::MessagePack.Formatters.InterfaceDictionaryFormatter<TREPLACE>()"},
+            {"System.Collections.Generic.IReadOnlyDictionary<,>", "global::MessagePack.Formatters.InterfaceReadOnlyDictionaryFormatter<TREPLACE>()"},
+            {"System.Collections.Generic.ICollection<>", "global::MessagePack.Formatters.InterfaceCollectionFormatter2<TREPLACE>()" },
+            {"System.Collections.Generic.IReadOnlyCollection<>", "global::MessagePack.Formatters.InterfaceReadOnlyCollectionFormatter<TREPLACE>()" },
+            {"System.Collections.Generic.IEnumerable<>", "global::MessagePack.Formatters.InterfaceEnumerableFormatter<TREPLACE>()" },
+            {"System.Linq.ILookup<,>", "global::MessagePack.Formatters.InterfaceLookupFormatter<TREPLACE>()" },
+            {"System.Linq.IGrouping<,>", "global::MessagePack.Formatters.InterfaceGroupingFormatter<TREPLACE>()" },
         };
 
         static void ExtractResolverInfo(InterfaceDefinition[] definitions, out GenericSerializationInfo[] genericInfoResults, out EnumSerializationInfo[] enumInfoResults, string messagePackGeneratedNamespace)

--- a/tests/MagicOnion.Generator.Tests/GenerateGenericsTest.cs
+++ b/tests/MagicOnion.Generator.Tests/GenerateGenericsTest.cs
@@ -941,5 +941,77 @@ namespace TempProject
                 "global::MessagePack.Formatters.ArrayFormatter<global::TempProject.MyResponse>"
             });
         }
+
+        
+        [Fact]
+        public async Task KnownFormatters()
+        {
+            using var tempWorkspace = TemporaryProjectWorkarea.Create();
+            tempWorkspace.AddFileToProject("IMyService.cs", @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MessagePack;
+using MagicOnion;
+
+namespace TempProject
+{
+    public interface IMyService : IService<IMyService>
+    {
+        UnaryResult<List<MyResponse>>                        MethodList(List<int> args);
+        UnaryResult<IList<MyResponse>>                       MethodIList();
+        UnaryResult<IReadOnlyList<MyResponse>>               MethodIROList();
+
+        UnaryResult<Dictionary<string, MyResponse>>          MethodDictionary();
+        UnaryResult<IDictionary<string, MyResponse>>         MethodIDictionary();
+        UnaryResult<IReadOnlyDictionary<string, MyResponse>> MethodIRODictionary();
+
+        UnaryResult<IEnumerable<MyResponse>>                 MethodIEnumerable();
+        UnaryResult<ICollection<MyResponse>>                 MethodICollection();
+        UnaryResult<IReadOnlyCollection<MyResponse>>         MethodIROCollection();
+
+        UnaryResult<ILookup<int, MyResponse>>                MethodILookup();
+        UnaryResult<IGrouping<int, MyResponse>>              MethodIGrouping();
+    }
+    public class MyResponse
+    {
+    }
+}
+            ");
+
+            var compiler = new MagicOnionCompiler(_testOutputHelper.WriteLine, CancellationToken.None);
+            await compiler.GenerateFileAsync(
+                tempWorkspace.CsProjectPath,
+                tempWorkspace.OutputDirectory,
+                true,
+                "TempProject.Generated",
+                "",
+                "MessagePack.Formatters"
+            );
+
+            var compilation = tempWorkspace.GetOutputCompilation();
+            compilation.GetCompilationErrors().Should().BeEmpty();
+            var symbols = compilation.GetNamedTypeSymbolsFromGenerated();
+            compilation.GetResolverKnownFormatterTypes().Should().Contain(new[]
+            {
+                "global::MessagePack.Formatters.ListFormatter<int>",
+                "global::MessagePack.Formatters.ListFormatter<global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceListFormatter2<global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceReadOnlyListFormatter<global::TempProject.MyResponse>",
+
+                "global::MessagePack.Formatters.DictionaryFormatter<string, global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceDictionaryFormatter<string, global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceReadOnlyDictionaryFormatter<string, global::TempProject.MyResponse>",
+
+                "global::MessagePack.Formatters.InterfaceEnumerableFormatter<global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceCollectionFormatter2<global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceReadOnlyCollectionFormatter<global::TempProject.MyResponse>",
+
+                "global::MessagePack.Formatters.InterfaceLookupFormatter<int, global::TempProject.MyResponse>",
+                "global::MessagePack.Formatters.InterfaceGroupingFormatter<int, global::TempProject.MyResponse>",
+
+            });
+        }
     }
 }

